From 32c529e3f76054a6372a5fcbde48b7059d542cc3 Mon Sep 17 00:00:00 2001
From: Tong Tiangen <tongtiangen@huawei.com>
Date: Tue, 9 Aug 2022 07:01:01 +0000
Subject: [PATCH 4417/4585] [Huawei] arm64/__mc_ex_table: fix compile error
 when CONFIG_ARCH_HAS_COPY_MC=n

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 187420

--------------------------------

There is an compile error when CONFIG_ARCH_HAS_COPY_MC is not set, fix it.

Fixes: 4ae0cf0689f1 ("[Huawei] arm64: add support for machine check error safe")
Signed-off-by: Tong Tiangen <tongtiangen@huawei.com>
---
 include/linux/extable.h | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/include/linux/extable.h b/include/linux/extable.h
index a5cdffabe89d..e608f8a8df4e 100644
--- a/include/linux/extable.h
+++ b/include/linux/extable.h
@@ -21,6 +21,12 @@ void trim_init_extable(struct module *m);
 const struct exception_table_entry *search_exception_tables(unsigned long add);
 #ifdef CONFIG_ARCH_HAS_MC_EXTABLE
 const struct exception_table_entry *search_mc_exception_tables(unsigned long add);
+#else
+static inline const struct exception_table_entry *
+search_mc_exception_tables(unsigned long add)
+{
+	return NULL;
+}
 #endif
 const struct exception_table_entry *
 search_kernel_exception_table(unsigned long addr);
@@ -30,6 +36,12 @@ search_kernel_exception_table(unsigned long addr);
 const struct exception_table_entry *search_module_extables(unsigned long addr);
 #ifdef CONFIG_ARCH_HAS_MC_EXTABLE
 const struct exception_table_entry *search_module_mc_extables(unsigned long addr);
+#else
+static inline const struct exception_table_entry *
+search_module_mc_extables(unsigned long addr)
+{
+	return NULL;
+}
 #endif
 #else
 static inline const struct exception_table_entry *
@@ -37,6 +49,11 @@ search_module_extables(unsigned long addr)
 {
 	return NULL;
 }
+static inline const struct exception_table_entry *
+search_module_mc_extables(unsigned long addr)
+{
+	return NULL;
+}
 #endif /*CONFIG_MODULES*/
 
 #ifdef CONFIG_BPF_JIT
-- 
2.17.1

